File: 🟡 Medium/53 - Maximum Subarray/_53MaximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        // KADANE'S ALGORITHM

        int cs = 0;
        int maxSum = Integer.MIN_VALUE;

        for (int num : nums) {
            cs += num;
            maxSum = Math.max(maxSum, cs);

            if (cs < 0) {
                cs = 0;
            }
        }

        return maxSum;
    }
}

File: 🟡 Medium/53 - Maximum Subarray/_53MaximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # KADANE'S ALGORITHM
        cs = 0
        maxSum = min(nums)

        for i in range(0, len(nums)):
            cs += nums[i]
            maxSum = max(maxSum, cs)

            if cs < 0:
                cs = 0

        return maxSum
